caffe2/core/tensor.h

```diff
@@ -608,27 +638,36 @@ class CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
            storage_.capacity() >= numel_ * storage_.itemsize())) {
         return storage_.data();
       }
+      const at::Allocator* allocator = storage_.allocator();
+      CAFFE_ENFORCE(allocator == nullptr);
       if (meta.ctor()) {
         // For types that need placement new, we will call it, as well as
         // making sure that when the data is freed, it calls the right
         // destruction procedure.
         auto size = numel_;
         auto dtor = storage_.dtype().dtor();
+        void* ptr;
+        at::DeleterFnPtr deleter;
         auto ptr_and_deleter =
```

Comment:
Can we easily make this function return a `DataPtr`? That will help out the code below a bit.